lib/criteo-api-retail.d.ts

```diff
@@ -0,0 +1,172 @@
+declare module 'criteo-api-retail';
+
+interface MetaData {
+    totalItemsAcrossAllPages: number,
+    currentPageSize: number,
+    currentPageIndex: number,
+    totalPages: number,
+    nextPage?: string
+}
+
+interface PageInfo {
+    pageIndex: number,
+    pageSize: number
+}
+
+interface AccountsResponse {
+    data: [
+        {
+            id: string,
+            type: string,
+            attributes: {
+                name: string
+                type: string,
+                subtype: string
+                countries: [
+                    string
+                ],
+                currency: string,
+                parentAccountLabel: string,
+                timeZone: string
+            }
+        }
+    ],
+    metadata: MetaData
+}
```

Comment:
```suggestion
interface Response<T> {
  data: T;
}

interface PagedResponse<T> extends Response<T> {
  metadata: MetaData;
}

interface AccountData {
  id: string,
  type: string,
  attributes: {
    name: string;
    type: string;
    subtype: string;
    countries: string[];
    currency: string;
    parentAccountLabel: string;
    timeZone: string;
  }
}
  
type AccountsResponse = PagedResponse<AccountData[]>
```
This can also be written as `interface AccountsResponse extends Response<AccountData[]> {}`
All depends on how you feel on types vs interfaces.